packages/engine/lib/memory/src/arrow/mod.rs

```diff
@@ -1,4 +1,5 @@
 //! TODO: DOC
+pub mod field;
```

Comment:
I don't think our Fields and FieldSpecs are strictly memory related things, they're abstractions at a higher level, and also aren't strictly Arrow. As we've just said on a call off-github, lets move them to their own crate for now and then hopefully group them with other stateful things later on like State, Context, etc.

Reply:
I moved it to the newly added `stateful` crate